databuilder/databuilder/extractor/delta_lake_metadata_extractor.py

```diff
@@ -311,6 +333,53 @@ def fetch_columns(self, schema: str, table: str) -> List[ScrapedColumnMetadata]:
                     parsed_columns[row['col_name']].set_is_partition(True)
         return list(parsed_columns.values())
 
+    def _iterate_nested_columns(self,
+                                parent: str,
+                                curr_field: Any,
+                                parsed_columns: Dict,
+                                total_cols: int) -> int:
+        if len(parent) > 0:
+            col_name = f"{parent}.{curr_field.name}"
+        else:
+            col_name = curr_field.name
+
+        if self.is_complex_delta_type(curr_field.dataType):
+            parsed_columns[col_name] = ScrapedColumnMetadata(
+                name=col_name,
+                data_type=curr_field.dataType.simpleString(),
+                sort_order=total_cols,
+                description=None,
```

Comment:
assume no description for nested file?

Reply:
At first pass I was under the impression that you couldn't. I am going to double check to be sure.

Reply:
Interesting. This works for DDL:
```sql
CREATE TABLE playground.jack_delta_typing_test (struct_col struct<a:string COMMENT "NESTED STRING COL", b:int COMMENT "NESTED INT COL"> COMMENT "THIS IS A STRUCT COL") USING delta;
```
but I'm unsure how to extract the nested comment:
```python
from pprint import pprint
raw = spark.sql("DESCRIBE playground.jack_delta_typing_test").collect()
pprint(raw)
```
Output:
```
[Row(col_name='struct_col', data_type='struct<a:string,b:int>', comment='THIS IS A STRUCT COL'),
 Row(col_name='', data_type='', comment=''),
 Row(col_name='# Partitioning', data_type='', comment=''),
 Row(col_name='Not partitioned', data_type='', comment='')]
```

Reply:
@feng-tao I updated the commit message with this as a design decision. I am also going to file a Support Ticket with Databricks about this to see if I am missing something or if it is a current limitation. This PR is ready for another pass.